src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/WcfTestCase.cs

```diff
@@ -40,11 +43,41 @@ internal WcfTestCase(IXunitTestCase testCase, string skippedReason)
 
         public void Deserialize(IXunitSerializationInfo info) { _testCase.Deserialize(info); }
 
-        public Task<RunSummary> RunAsync(
+        public async Task<RunSummary> RunAsync(
             IMessageSink diagnosticMessageSink, IMessageBus messageBus, object[] constructorArguments,
             ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource)
         {
-            return new XunitTestCaseRunner(this, DisplayName, _skippedReason, constructorArguments, TestMethodArguments, messageBus, aggregator, cancellationTokenSource).RunAsync();
+            ConcurrentQueue<EventWrittenEventArgs> events = new ConcurrentQueue<EventWrittenEventArgs>();
+	          s_testListener._eventWritten = events.Enqueue;
+
+            var runsummary = await (new XunitTestCaseRunner(this, DisplayName, _testCase.SkipReason, constructorArguments, TestMethodArguments, messageBus, aggregator, cancellationTokenSource)).RunAsync();
+
+            if (runsummary.Failed > 0)
```

Comment:
Be aware that Theory based tests can run multiple tests under this 'await' call (which is why Failed is an int).  It may not matter, but this new ETW logic will fire if any of the multiple tests fail, even if some of them succeed.  I point that out so you know, but we probably can ignore that for now.